### PR TITLE
Add configurable capacity map

### LINKED ISSUE #38: Hard coded capacity map
We currently use a hard-coded dictionary to choose the size of the model. Need to make this modifiable by the user.

https://github.com/ssg-research/amulet/blob/a1619c0dc729a4a4dee87c955c3deda21c629b73/amulet/utils/__pipeline.py#L121C1-L121C13

### PR DESCRIPTION
**Pull request by asim29**
Closes #38

### CODE REVIEW

File: amulet/utils/__pipeline.py [hunk 1]
```diff
@@ -118,12 +118,82 @@ def create_dir(path: Path | str, log: logging.Logger | None = None) -> Path:
     return resolved_path
 
 
-capacity_map = {
-    "m1": {"vgg": "VGG11", "linearnet": [128, 256, 128]},
-    "m2": {"vgg": "VGG13", "linearnet": [256, 512, 256]},
-    "m3": {"vgg": "VGG16", "linearnet": [512, 1024, 512]},
+DEFAULT_CAPACITY_MAP = {
```
Comment (sebszyller) [inline amulet/utils/__pipeline.py:121]: why mixed int and str?

**Review by sebszyller — COMMENTED** [review]
Btw, we'd remove all models later; and keep only ones that we use for examples.

File: amulet/utils/__pipeline.py (hunk 1 above)
Comment (asim29) [reply amulet/utils/__pipeline.py:121]: the str is used for MaxPool layers (or theoretically other non-standard layers)

**Comment by asim29** [comment]
Once this is approved I can also merge this.

**Review by sebszyller — APPROVED** [review]

**Pull request merged**